src/services/TestResultsService.ts

```diff
@@ -302,13 +302,13 @@ export class TestResultsService {
                     const firstTestAfterAnvCompareDate = dateFns.addYears(dateFns.startOfMonth(regOrFirstUseDate!), 1);
                     // Checks for testType = First test or First test Retest AND test date is 1 year from the month of first use or registration date
                     if (this.isFirstTestRetestTestType(testType) && dateFns.isAfter(new Date(), firstTestAfterAnvCompareDate)) {
-                        testType.testExpiryDate = dateFns.addYears(dateFns.lastDayOfMonth(new Date()), 1).toISOString();
+                      testType.testExpiryDate = dateFns.lastDayOfMonth(dateFns.addYears(new Date(), 1)).toISOString();
```

Comment:
Can we wrap this up in a function as we are repeating the logic in 4 different places. Something like `getLastDayofMonth(date)`

Reply:
done